glue/viewers/image/composite_array.py

```diff
@@ -104,27 +143,45 @@ def __call__(self, bounds=None):
             else:
                 scalar = False
 
-            data = STRETCHES[layer['stretch']]()(contrast_bias(interval(array)))
+            data = interval(array)
+            data = contrast_bias(data, out=data)
+            data = stretch(data, out=data)
             data[np.isnan(data)] = 0
 
-            if isinstance(layer['color'], Colormap):
+            if self.mode == 'colormap':
 
                 if img is None:
                     img = np.ones(data.shape + (4,))
 
                 # Compute colormapped image
-                plane = layer['color'](data)
+                plane = layer['cmap'](data)
+
+                # Check what the smallest colormap alpha value for this layer is
+                # - if it is 1 then this colormap does not change transparency,
+                # and this allows us to speed things up a little.
+
+                if layer['cmap'](CMAP_SAMPLING)[:, 3].min() == 1:
+
+                    if layer['alpha'] == 1:
+                        img[...] = 0
+                    else:
+                        plane *= layer['alpha']
+                        img *= (1 - layer['alpha'])
+
+                else:
+
+                    # Use traditional alpha compositing
+
+                    alpha_plane = layer['alpha'] * plane[:, :, 3]
 
-                alpha_plane = layer['alpha'] * plane[:, :, 3]
+                    plane[:, :, 0] = plane[:, :, 0] * alpha_plane
+                    plane[:, :, 1] = plane[:, :, 1] * alpha_plane
+                    plane[:, :, 2] = plane[:, :, 2] * alpha_plane
```

Comment:
```suggestion
                    plane[:, :, :3] *= np.atleast3d(alpha_plane)
```
Probably not necessarily faster with the bit of extra vectorisation – doing the channels separately could in fact be more memory-efficient.